test/evp_libctx_test.c

```diff
@@ -364,6 +369,11 @@ static int test_cipher_reinit(int test_id)
     /* DES3-WRAP uses random every update - so it will give a different value */
     diff = EVP_CIPHER_is_a(cipher, "DES3-WRAP");
 
+    /* eNULL-HMAC- use implicit fetching of digest algo */
+    if (EVP_CIPHER_is_a(cipher, "eNULL-HMAC-SHA256")
+        || EVP_CIPHER_is_a(cipher, "eNULL-HMAC-SHA384"))
+        prev_libctx = OSSL_LIB_CTX_set0_default(libctx);
```

Comment:
Why do you need to do this?

Reply:
The code needs to work without this. Provided ciphers are expected to use the correct libctx. You should instead use the newer EVP_MAC APIs which enable you to use the right libctx.

Reply:
I could not find any where `libctx` is exposed to cipher implementation.

Reply:
> I could not find any where `libctx` is exposed to cipher implementation.

Please look at the other ciphers - the provctx argument is used to get to the libctx.